lib/view/formulier/InvoerVelden.class.php

```diff
@@ -796,54 +796,98 @@ class RequiredLidField extends LidField {
 
 /**
  * Select an entity based on primary key values in hidden input fields, supplied by remote data source.
+ *
+ * NOTE: support alleen entities met een enkele primary key.
  */
 class EntityField extends InputField {
 
-	private $show_value;
-
-	public function __construct($name, array $primary_key_values, $description, PersistenceModel $model, $show_value, $find_url = null) {
-		parent::__construct($name, $primary_key_values, $description, $model);
-		$this->show_value = $show_value;
-		$this->suggestions[] = $find_url;
-	}
-
-	public function validate() {
+	private $show_value, $entity, $orm;
+
+    /**
+     * @var PersistenceModel
+     */
+    protected $model;
+
+    /**
+     * EntityField constructor.
+     * @param $name string Prefix van de input
+     * @param $show string Attribuut van $model om in de input weer te geven
+     * @param $description string Beschrijvijng van de input
+     * @param PersistentEntity $entity
+     * @param PersistenceModel $model Model
+     * @param $url string Url waar aanvullingen te vinden zijn
+     */
+    public function __construct($name, $show, $description, PersistenceModel $model, $url) {
+        $this->orm = $model::orm;
+        $this->entity = new $this->orm();
+        foreach ($this->entity->getPrimaryKey() as $key) {
+            $this->entity->$key = filter_input(INPUT_POST, $name . '_'. $key, FILTER_DEFAULT);
+        }
+
+        parent::__construct($name, $this->entity->$show, $description, $model);
+        $this->suggestions[] = $url;
+        $this->show_value = $show;
+    }
+
+    public function getName() {
+        // TODO: Dit moet de volledige pk kunnen zijn.
+        return $this->name . '_' . $this->entity->getPrimaryKey()[0];
+    }
+
+    public function getValue() {
+        // TODO: Dit moet de volledige pk kunnen zijn.
+        $key = $this->entity->getPrimaryKey()[0];
+        return $this->entity->$key;
+    }
+
+    public function validate() {
 		if (!parent::validate()) {
 			return false;
 		}
 		// parent checks not null
 		if ($this->value == '') {
 			return true;
 		}
-		// bestaat er een entity met opgegeven primary key?
-		$where = array();
-		$class = $this->model->orm;
-		$orm = new $class();
-		foreach ($orm->getPrimaryKey() as $key) {
-			$where[] = $key . ' = ?';
-		}
-		if (!$this->model->exist(implode(' AND ', $where), $this->value)) {
```

Comment:
> Als de entity uit het EntityField een composite primary key heeft, moet de entity van Formulier een composite foreign key hebben, dit werkt nog niet in Formulier.

Exact :-) Dit is precies waarom EntityField niet is afgemaakt door mij, want het framework heeft nog geen ondersteuning voor foreign keys, laat staan composite foreign keys. Ik zit nu dus af te wegen of we gewoon geen composite keys meer willen (gebruikelijk en mijn voorkeur) of UUID telkens encoden/decoden.

FK support is erg eenvoudig toe te voegen, het is dus vooral de design keuze op het punt van composite keys. Wat vind jij @qurben ?